### PR TITLE
issue/cluster-postgres-validation-coverage

### LINKED ISSUE #495: Clustering H07: enforce PostgreSQL integration inventory and CI coverage
Refs #493, H07. Design: PR #494, [H07](https://github.com/bunny-lab-io/Borealis/blob/7e09cf9e81d26cc18a42739b458be6df0751dd36/Docs/Reference/Future_Roadmaps/engine-clustering-mvp-readiness-and-remote-onboarding.md#h07).

ISSUE GOAL:
- Expected behavior: PostgreSQL lane inventories and executes every required database-backed Go integration test, including four currently omitted maintenance/completion/retry/action-image tests. Missing tests, unregistered database tests, required skips, and zero-test selections fail. Ordinary cluster/controller/store/scheduler changes select database CI.
- Affected subsystem: Tests database runner, maintained manifests, portable policy and CI path selection.
- Relevant docs: Docs/Reference/Unit_Testing.md, Docs/Reference/testing-regressions.md, Docs/Engine/managing-engine-clusters.md, Docs/Reference/Data and Schema/db-reference.md; PR #494 H07.
- Validation: repository policy; focused inventory/selection/result-audit unit tests; Tests/run-database-postgres.sh against isolated PostgreSQL 17; affected Engine Go and documentation checks.
- Docs/SBOM/debt impact: update canonical testing and regression guidance. No runtime dependency or schema change planned. Preserve #460/#463 guards.

Owner: @nicole-rappe; implementer: Codex in current operator-authorized session. Branch/PR: issue/cluster-postgres-validation-coverage. No prerequisite. H01 depends on this coverage; H01 fencing behavior/tests remain separately reviewable. Later H02-H06 behavioral regressions land with their owning children and register in this inventory when database-backed. Q01 live qualification remains #480/#482.

Done: inventory drift/skip failure behavior is tested, all eight current PostgreSQL tests execute with retained results, cluster Go changes select database gate, docs match, PR validation recorded. Keep #493 open; this child does not certify live clustering.

Implementation ready for review in [PR #497](https://github.com/bunny-lab-io/Borealis/pull/497). [Validation and resumable handover](https://github.com/bunny-lab-io/Borealis/issues/493#issuecomment-5553519225); merge order H07 #497 then H01 #498 after main retarget and revalidation. Umbrella remains open.

### PR DESCRIPTION
**Pull request by nicole-rappe**
Four PostgreSQL regressions were absent from database runner, and ordinary cluster/controller/store/scheduler edits did not select database CI. Maintained inventory now covers all eight database-gated Go tests. Source discovery binds package helper declarations, handles cross-file references and local shadowing, and prevents same-named methods or external test packages hiding required tests. Runner rejects skipped, missing, failed or incomplete results.

Refs #493 (H07). Closes #495. H01 #496 / PR #498 depends on this foundation. Umbrella remains open through implementation and #480/#482 exact-release qualification.

- Database CI selects API Go source/tests, internal helpers and module changes; CI retains per-test JSON and logs.
- Canonical testing guidance and REG-TEST-050 document inventory and audit requirements. No dependency, schema or runtime behavior change.

Validation at `535d9ec01dce0ece5e79b986d42878b654be33c5`:
- `PYTHONDONTWRITEBYTECODE=1 python3 -m unittest discover -s Tests/Unit_Tests -p 'test_*.py'`: PASS, 81 tests including 8 PostgreSQL inventory/result/selection regressions.
- `BOREALIS_DOCKER_USE_SUDO=1 BOREALIS_ENGINE_TEST_PYTHON=/opt/Borealis/Unit_Test_Results/database-postgres-20260905T164144Z/venv/bin/python bash Tests/run-database-postgres.sh`: PASS, isolated PostgreSQL 17 Python contracts plus all 8 required Go tests, zero skips. Results `Unit_Test_Results/database-postgres-20260905T174200Z/`.
- `bash Tests/run-docs.sh`: PASS, strict build; results `Unit_Test_Results/docs-20260905T174200Z/`. Inventory and `git diff --check`: PASS.
- `bash Tests/run-repository-policy.sh`: BLOCKED locally, exit 127, missing `node`. Fresh [PR validation](https://github.com/bunny-lab-io/Borealis/actions/runs/33981795017) and [CodeQL](https://github.com/bunny-lab-io/Borealis/actions/runs/33981793056) both PASS on exact head. [Database artifact](https://github.com/bunny-lab-io/Borealis/actions/runs/33981795017/artifacts/9973979587) independently downloaded and audited: all 8 required Go tests, zero skips.
- [Review finding](https://github.com/bunny-lab-io/Borealis/pull/497#discussion_r3941420467) fixed with collision/cross-file/shadow/external-package regression and resolved.

[Previous full implementation evidence](https://github.com/bunny-lab-io/Borealis/issues/493#issuecomment-5553519225); [approved merge sequence and review correction](https://github.com/bunny-lab-io/Borealis/issues/493#issuecomment-5553608795). No live Engine database, K3s membership or deployment changes; disposable database container cleaned.

Merged with operator approval. [Final merge evidence, cleanup and resumable handover](https://github.com/bunny-lab-io/Borealis/issues/493#issuecomment-5553674517). Umbrella #493 remains open.

### CODE REVIEW

File: Tests/tools/postgresinventory/main.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	files, err := filepath.Glob(filepath.Join(*root, "*_test.go"))
+	if err != nil || *root == "" || len(files) == 0 {
+		fmt.Fprintln(os.Stderr, "POSTGRES INVENTORY FAIL: test source directory missing or empty")
+		os.Exit(1)
+	}
+	functions := map[string]*ast.FuncDecl{}
+	for _, path := range files {
+		file, err := parser.ParseFile(token.NewFileSet(), path, nil, 0)
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
```
[the remark this answers is not in the document]
Comment (nicole-rappe) [reply Tests/tools/postgresinventory/main.go:33]: Fixed in `535d9ec01dce0ece5e79b986d42878b654be33c5`. Discovery keys database state by function declaration, separates package scopes, excludes receiver methods from package helper lookup, and honors parser-resolved local bindings. Selector names cannot resolve as package helpers.

Regression fixture contains same-named package helper/method, indirect caller, cross-file caller, local shadow and external test package. Correct two database callers remain discoverable. `python3 -m unittest Tests.Unit_Tests.test_postgres_inventory`: 8 passed; full repository test set: 81 passed. Isolated PostgreSQL lane: all 8 required Go tests executed, zero skips; strict docs build passed. Fresh GitHub checks pending before approved merge.
